lib/src/clojure_lsp/feature/add_missing_libspec.clj

```diff
@@ -175,61 +224,112 @@
-                         (let [existing-class-name (symbol (last (string/split (z/string exisiting-import-package-by-single-full-package) #"\.")))]
+            result-loc (or result-loc ;if not modified lib add new entry
                            (z/subedit-> ns-zip
-                                        (z/find-next z/next #(= exisiting-import-package-by-single-full-package %))
-                                        (z/replace [lib-sym existing-class-name class-sym])))
-
-                         :else
-                         (z/subedit-> ns-zip
-                                      (cond->
-                                       add-form-type? (z/append-child (n/newlines 1))
-                                       add-form-type? (z/append-child (n/spaces 2))
-                                       add-form-type? (z/append-child (list libspec-type)))
-                                      (z/find-value z/next libspec-type)
-                                      (z/up)
-                                      (cond->
-                                       (or (not add-form-type?)
-                                           (= :next-line ns-inner-blocks-indentation)) (z/append-child* (n/newlines 1)))
-                                      (z/append-child* (n/spaces (dec col)))
-                                      (z/append-child form-to-add)))]
+                                        (cond->
+                                         add-form-type? (z/append-child (n/newlines 1))
+                                         add-form-type? (z/append-child (n/spaces 2))
+                                         add-form-type? (z/append-child (list libspec-type)))
+                                        (z/find-value z/next libspec-type)
+                                        (z/up)
+                                        (cond->
+                                         (or (not add-form-type?)
+                                             (= :next-line ns-inner-blocks-indentation)) (z/append-child* (n/newlines 1)))
+                                        (z/append-child* (n/spaces (dec col)))
+                                        (z/append-child form-to-add)))]
         [{:range (meta (z/node result-loc))
           :loc result-loc}]))))
 
+(defn ^:private add-to-rcf?
+  "returns zloc of rcf when require should be done in rcf-form, otherwise nil"
+  [zloc db producer]
+  (let [rcf-zloc (edit/inside-rcf? zloc)
+        c-setting (settings/get db [:completion :add-libs-inside-rcf])]
+    (when (and
+            (not (= false? c-setting))
+            rcf-zloc
+            (or c-setting
+                (and
+                  producer
+                  (= "Yes"
+                     (producer/show-message-request
+                       producer
+                       "Add require inside this comment form? (to avoid this prompt, set { :completion {:add-libs-inside-rcf true|false }.)"
```

Comment:
I'd leave this as:
```suggestion
                       "Add require inside this comment form?"
```

And if anyone complains we can point to the docs

Reply:
thanks for your quick response - sure, that's fine with me, I'll update code & docs (forgot to push them yesterday...)
I wrote the prompt before adding the functionality for imports, maybe we should ask "Add lib inside this comment form?" instead?

Reply:
Nevermind, I can just parametrize it and ask for require and import respectively.